src/lib/route_rule.rs

```diff
@@ -0,0 +1,211 @@
+    }
+    let mut links = handle.rule().get(IpVersion::V4).execute();
+    while let Some(rt_msg) = links.try_next().await? {
+        rules.push(get_rule(rt_msg));
+    }
+    Ok(rules)
+}
+
+fn get_rule(rule_msg: RuleMessage) -> RouteRule {
+    let mut rl = RouteRule::default();
+    let header = &rule_msg.header;
+    rl.address_family = header.family.into();
+    let src_prefix_len = header.src_len;
+    let dst_prefix_len = header.dst_len;
+    rl.tos = header.tos;
+    rl.action = header.action.into();
+    if header.table > RT_TABLE_UNSPEC {
+        rl.table = Some(header.table.into());
+    }
+    let family = &rl.address_family;
+    for nla in &rule_msg.nlas {
+        match nla {
+            Nla::Destination(ref d) => {
+                rl.dst = Some(format!(
+                    "{}/{}",
+                    _addr_to_string(d, family),
+                    dst_prefix_len,
+                ));
+            }
+            Nla::Source(ref d) => {
+                rl.src = Some(format!(
+                    "{}/{}",
+                    _addr_to_string(d, family),
+                    src_prefix_len,
+                ));
+            }
+            Nla::Iifname(ref d) => {
+                rl.iif = Some(d.clone().to_string());
+            }
+            Nla::OifName(ref d) => {
+                rl.oif = Some(d.clone().to_string());
+            }
+            Nla::Goto(ref d) => {
+                rl.goto = Some(*d);
+            }
+            Nla::Priority(ref d) => {
+                rl.priority = Some(*d);
+            }
+            Nla::FwMark(ref d) => {
+                rl.fw_mark = Some(*d);
+            }
+            Nla::FwMask(ref d) => {
+                rl.fw_mask = Some(*d);
+            }
+            Nla::Flow(ref d) => {
+                rl.flow = Some(*d);
+            }
+            Nla::TunId(ref d) => {
+                rl.tun_id = Some(*d);
+            }
+            Nla::SuppressIfGroup(ref d) => {
+                if *d != std::u32::MAX {
+                    rl.suppress_ifgroup = Some(*d);
+                }
+            }
+            Nla::SuppressPrefixLen(ref d) => {
+                if *d != std::u32::MAX {
+                    rl.suppress_prefix_len = Some(*d);
+                }
+            }
+            Nla::Table(ref d) => {
+                if *d > RT_TABLE_UNSPEC.into() {
+                    rl.table = Some(*d);
+                }
+            }
+            Nla::Protocol(ref d) => {
+                rl.protocol = Some(d.clone().into());
+            }
+            Nla::IpProto(ref d) => {
+                rl.ip_proto = Some(d.clone().into());
+            }
+            _ => eprintln!("Unknown NLA message for roule {:?}", nla),
+        }
+    }
+
+    rl
+}
+
+fn _addr_to_string(data: &[u8], family: &AddressFamily) -> String {
+    match family {
+        AddressFamily::IPv4 => parse_as_ipv4(data),
+        AddressFamily::IPv6 => parse_as_ipv6(data),
+        _ => format!("{:?}", data),
```

Comment:
Instead of putting junk string there, please use `eprintln!()` instead.

Reply:
Handeling error now.